design/specs/add-formula-column.md

```diff
@@ -0,0 +1,180 @@
+
+![](/assets/design/specs/add-formula-column/6.png)
+
+#### Scenario 1.b - Add a text type formula column
+
+The text type formula works on columns with text data type. There can be different parameters depending on what the formula is. For instance, there are `starting index` and `count` here in the example below.
+
+![](/assets/design/specs/add-formula-column/7.png)
+
+#### Scenario 1.c - Add a number type formula column
+
+The number type formula works on columns with number data type. There can be different parameters depending on what the formula is. For instance, there is `comparison sign` here in the example below.
+
+![](/assets/design/specs/add-formula-column/8.png)
+
+#### Scenario 1.d - Add a date & time type formula column
+
+The date/time type formula works on columns with date/time data type. There can be different parameters depending on what the formula is. For instance, there is `precision` here in the example below.
+
+![](/assets/design/specs/add-formula-column/13.png)
+
+#### Scenario 1.e - Add a cumulative type formula column
+
+The cumulative type formula works on columns with any data type. There can be different parameters depending on what the formula is. It basically shows how data in one column is changing with the help of different mathematical equations.
+
+![](/assets/design/specs/add-formula-column/10.png)
+
+#### Scenario 1.f - Add a boolean type formula column
+
+The boolean type formula works on columns with boolean data type. There can be different parameters depending on what the formula is. For instance, there is `condition` here in the example below.
+
+![](/assets/design/specs/add-formula-column/9.png)
+
+#### Scenario 1.g - Add a regular expression type formula column
+
+The regular expression type formula works on columns with any data type. There can be different parameters depending on what the formula is. It basically is to determine if there are any patterns in the data of a column. 
+
+![](/assets/design/specs/add-formula-column/11.png)
+
+#### Scenario 1.h - Add a list type formula column
+
+The list type formula works on columns with list data type. There can be different parameters depending on what the formula is.
+
+![](/assets/design/specs/add-formula-column/12.png)
+
+### Scenario 2 - Adding an erroneous parameter to the formula 
+
+Parameters of different formulas accept columns of different data types and values of different data types. The erroneous parameter here means the data type of the entered value or column is incompatible with the formula. 
+
+For instance, the length formula expects a list value or a list type column as an input to the parameter `List` but it gets a text type column which is movieNames and hence throws an error and the user is not able to see a filled column.
+
```

Comment:
Instead of allowing users to select an invalid column, I believe we should prevent the error. It's worth noting that we limit the number of columns and formulas that can be chosen based on existing parameters.

Reply:
I agree with you limiting column options but the users can also add values to the parameters. 

Plus I don't think we should limit the formula list based on the parameters since it might restrict user in going to a completely different formula if they are not happy with the result. 

LMK what you think